File: src/main/java/com/flower/game/progress/controller/PlayProgressController.java
package com.flower.game.progress.controller;

import com.flower.game.progress.model.dto.DailyInfoComputeRequest;
import com.flower.game.progress.model.dto.PlayProgressQueryRequest;
import com.flower.game.progress.model.dto.PlayProgressSaveRequest;
import com.flower.game.progress.model.dto.SpiritualRepoQueryRequest;
import com.flower.game.progress.model.entity.PlayProgress;
import com.flower.game.progress.model.vo.PlayProgressVO;
import com.flower.game.progress.model.vo.SpiritualRepoInfoVO;
import com.flower.game.progress.service.GamePlayProgressService;
import com.flower.game.progress.service.IPlayProgressService;
import common.annotations.ApiErrorCode;
import common.baseEntities.BaseResponse;
import common.exceptions.ErrorCode;
import common.page.PageVO;
import common.utils.ResultUtils;
import common.utils.ThrowUtils;
import jakarta.annotation.Resource;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * <p>
 * 用户游玩进度表 前端控制器
 * </p>
 *
 * @author F1ower
 * @since 2026-01-10
 */
@RestController
@RequestMapping("/playProgress")
public class PlayProgressController {

    @Resource
    private IPlayProgressService IPlayProgressService;

    @Resource
    private GamePlayProgressService gamePlayProgressService;

    /**
     * 用户游玩进度保存
     *
     * @param request 保存请求
     * @return 自定义响应
     */
    @PostMapping("save/playProgress")
    @ApiErrorCode(ErrorCode.SAVE_ERROR)
    public BaseResponse<String> savePlayProgress(@RequestBody PlayProgressSaveRequest request) {
        IPlayProgressService.savePlayProgress(request);
        return ResultUtils.success("游戏进度保存成功");
    }

    /**
     * 用户游玩进度查询
     *
     * @param request 查询请求
     * @return 游玩进度
     */
    @PostMapping("query/playProgress")
    @ApiErrorCode(ErrorCode.QUERY_ERROR)
    public BaseResponse<PlayProgressVO> queryPlayProgress(@RequestBody PlayProgressQueryRequest request) {
        PlayProgressVO playProgressVO = IPlayProgressService.queryPlayProgress(request);
        return ResultUtils.success(playProgressVO);
    }


    /**
     * 计算当天的必要信息
     *
     * @param request 计算请求
     * @return 自定义响应
     */
    @PostMapping("compute/dailyInfo")
    @ApiErrorCode(ErrorCode.COMPUTE_ERROR)
    public BaseResponse<String> computeDailyInfo(@RequestBody DailyInfoComputeRequest request) {
        // 计算当亲的信息
        gamePlayProgressService.computeDailyRatio(request);
        return ResultUtils.success("每日信息计算完毕");
    }

    /**
     * 查询灵材仓库
     *
     * @param queryRequest 查询请求
     * @return 灵材仓库信息
     */
    @PostMapping("list/SpiritualRepo")
    @ApiErrorCode(ErrorCode.SPIRITUAL_REPO_QUERY_ERROR)
    public BaseResponse<PageVO<SpiritualRepoInfoVO>> listSpiritualRepoByPage(@RequestBody SpiritualRepoQueryRequest queryRequest) {
        // 校验参数
        ThrowUtils.throwIf(queryRequest == null, ErrorCode.PARAM_ERROR, "灵材仓库查询请求不能为空");
        PageVO<SpiritualRepoInfoVO> spiritualRepoInfoVOPageVO = gamePlayProgressService.listSpiritualRepoByPage(queryRequest);
        return ResultUtils.success(spiritualRepoInfoVOPageVO);
    }
}

File: ai-simulation-game-frontend/src/service/api/playProgressController.ts
// @ts-ignore
/* eslint-disable */
import request from "@/request.ts";

/** 此处后端没有提供注释 POST /playProgress/compute/dailyInfo */
export async function computeDailyInfo(
  body: API.DailyInfoComputeRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseString>("/playProgress/compute/dailyInfo", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /playProgress/list/SpiritualRepo */
export async function listSpiritualRepoByPage(
  body: API.SpiritualRepoQueryRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponsePageVOSpiritualRepoInfoVO>(
    "/playProgress/list/SpiritualRepo",
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /playProgress/query/playProgress */
export async function queryPlayProgress(
  body: API.PlayProgressQueryRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponsePlayProgressVO>(
    "/playProgress/query/playProgress",
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      data: body,
      ...(options || {}),
    }
  );
}

/** 此处后端没有提供注释 POST /playProgress/save/playProgress */
export async function savePlayProgress(
  body: API.PlayProgressSaveRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseString>("/playProgress/save/playProgress", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    data: body,
    ...(options || {}),
  });
}
